packages/waku/src/router/create-pages.ts

```diff
@@ -457,3 +471,357 @@ export function createPages<
+        const noSsr = noSsrSet.has(pathSpec);
+        const isStatic = (() => {
+          for (const [_, [layoutPathSpec]] of dynamicLayoutPathMap) {
+            if (hasPathSpecPrefix(layoutPathSpec, pathSpec)) {
+              return false;
+            }
+          }
+          return true;
+        })();
+
+        const pattern = path2regexp(parsePathWithSlug(path));
+
+        const layoutPaths = getLayouts(pattern);
+
+        paths.push({
+          pattern,
+          path: pathSpec,
+          components: {
+            ...layoutPaths.reduce<Record<string, { isStatic: boolean }>>(
+              (acc, lPath) => {
+                acc[`layout:${lPath}`] = {
+                  isStatic: !dynamicLayoutPathMap.has(lPath),
+                };
+                return acc;
+              },
+              {},
+            ),
+            root: { isStatic: rootIsStatic },
+            [`route:${path}`]: { isStatic },
+            [`page:${path}`]: { isStatic },
+          },
+          noSsr,
+        });
+      }
+      for (const [path, [pathSpec]] of dynamicPagePathMap) {
+        const noSsr = noSsrSet.has(pathSpec);
+        const pattern = path2regexp(parsePathWithSlug(path));
+        const layoutPaths = getLayouts(pattern);
+        paths.push({
+          pattern,
+          path: pathSpec,
+          components: {
+            ...layoutPaths.reduce<Record<string, { isStatic: boolean }>>(
+              (acc, lPath) => {
+                acc[`layout:${lPath}`] = {
+                  isStatic: !dynamicLayoutPathMap.has(lPath),
+                };
+                return acc;
+              },
+              {},
+            ),
+            root: { isStatic: rootIsStatic },
+            [`route:${path}`]: { isStatic: false },
+            [`page:${path}`]: { isStatic: false },
+          },
+          noSsr,
+        });
+      }
+      for (const [path, [pathSpec]] of wildcardPagePathMap) {
+        const noSsr = noSsrSet.has(pathSpec);
+        const pattern = path2regexp(parsePathWithSlug(path));
+        const layoutPaths = getLayouts(pattern);
+        paths.push({
+          pattern: path2regexp(parsePathWithSlug(path)),
+          path: pathSpec,
+          components: {
+            ...layoutPaths.reduce<Record<string, { isStatic: boolean }>>(
+              (acc, lPath) => {
+                acc[`layout:${lPath}`] = {
+                  isStatic: !dynamicLayoutPathMap.has(lPath),
+                };
+                return acc;
+              },
+              {},
+            ),
+            root: { isStatic: rootIsStatic },
+            [`route:${path}`]: { isStatic: false },
+            [`page:${path}`]: { isStatic: false },
+          },
+          noSsr,
+        });
+      }
+      return paths;
+    },
+    renderRoute: async (path, options) => {
+      await configure();
+
+      // Should skip be a Set<string>? https://github.com/dai-shi/waku/blob/main/examples/22_define-router/src/entries.tsx#L63
```

Comment:
It's coming from a serialized string, so `Array<string>` should be easier to deal with. (at least, for now)

Reply:
makes sense 👍 - wasn't sure if this crossed the client server boundary or not, so this helps my understanding, thanks!